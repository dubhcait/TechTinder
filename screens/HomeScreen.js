import * as WebBrowser from 'expo-web-browser';
import React from 'react';
import { StyleSheet, Text, View, Dimensions, Image, Animated, PanResponder } from 'react-native';

import NopeComponent from '../components/NopeStamp';
import LikeComponent from '../components/LikeStamp';
import EndComponent from '../components/EndMessage';

const SCREEN_HEIGHT = Dimensions.get('window').height
const SCREEN_WIDTH = Dimensions.get('window').width

const Tech = [
  { id: "1", uri: require('../assets/cards/React.png') },
  { id: "2", uri: require('../assets/cards/css.png') },
  { id: "3", uri: require('../assets/cards/elixir.jpeg') },
  { id: "4", uri: require('../assets/cards/js.png') },
  { id: "5", uri: require('../assets/cards/go.png') },
  { id: "6", uri: require('../assets/cards/html.png') },
  { id: "7", uri: require('../assets/cards/firebase.png') },
  { id: "8", uri: require('../assets/cards/svg.png') },
  { id: "9", uri: require('../assets/cards/python.jpeg') },

]

export default class HomeScreen extends React.Component {

  constructor() {
    super()

    this.position = new Animated.ValueXY()
    this.state = {
      currentIndex: 0
    }

    this.rotate = this.position.x.interpolate({
      inputRange: [-SCREEN_WIDTH / 2, 0, SCREEN_WIDTH / 2],
      outputRange: ['-10deg', '0deg', '10deg'],
      extrapolate: 'clamp'
    })

    this.rotateAndTranslate = {
      transform: [{
        rotate: this.rotate
      },
      ...this.position.getTranslateTransform()
      ]
    }

    this.likeOpacity = this.position.x.interpolate({
      inputRange: [-SCREEN_WIDTH / 2, 0, SCREEN_WIDTH / 2],
      outputRange: [0, 0, 1],
      extrapolate: 'clamp'
    })
    this.dislikeOpacity = this.position.x.interpolate({
      inputRange: [-SCREEN_WIDTH / 2, 0, SCREEN_WIDTH / 2],
      outputRange: [1, 0, 0],
      extrapolate: 'clamp'
    })

    this.nextCardOpacity = this.position.x.interpolate({
      inputRange: [-SCREEN_WIDTH / 2, 0, SCREEN_WIDTH / 2],
      outputRange: [1, 0, 1],
      extrapolate: 'clamp'
    })
    this.nextCardScale = this.position.x.interpolate({
      inputRange: [-SCREEN_WIDTH / 2, 0, SCREEN_WIDTH / 2],
      outputRange: [1, 0.7, 1],
      extrapolate: 'clamp'
    })

  }
  componentWillMount() {
    this.PanResponder = PanResponder.create({

      onStartShouldSetPanResponder: (evt, gestureState) => true,
      onPanResponderMove: (evt, gestureState) => {

        this.position.setValue({ x: gestureState.dx, y: gestureState.dy })
      },
      onPanResponderRelease: (evt, gestureState) => {

        if (gestureState.dx > 120) {
          Animated.spring(this.position, {
            toValue: { x: SCREEN_WIDTH + 100, y: gestureState.dy }
          }).start(() => {
            this.setState({ currentIndex: this.state.currentIndex + 1 }, () => {
              this.position.setValue({ x: 0, y: 0 })
            })
          })
        }
        else if (gestureState.dx < -120) {
          Animated.spring(this.position, {
            toValue: { x: -SCREEN_WIDTH - 100, y: gestureState.dy }
          }).start(() => {
            this.setState({ currentIndex: this.state.currentIndex + 1 }, () => {
              this.position.setValue({ x: 0, y: 0 })
            })
          })
        }
        else {
          Animated.spring(this.position, {
            toValue: { x: 0, y: 0 },
            friction: 4
          }).start()
        }
      }
    })
  }

  renderTech = () => {

    return Tech.map((item, i) => {


      if (i < this.state.currentIndex) {
        return null
      }
      else if (i == this.state.currentIndex) {

        return (
          <Animated.View
            {...this.PanResponder.panHandlers}
            key={item.id} style={[this.rotateAndTranslate, { height: SCREEN_HEIGHT - 120, width: SCREEN_WIDTH, padding: 20, position: 'absolute', ...styles.imageContainer }]}>
           
            <LikeComponent style={{ likeAnimation: { ...styles.likeAnimation, opacity: this.likeOpacity }, Like: styles.Like }} />
            <NopeComponent style={{ nopeAnimation: { ...styles.nopeAnimation, opacity: this.dislikeOpacity}, Nope: styles.Nope }} />


            <Image
              style={styles.image}
              source={item.uri} />

          </Animated.View>
        )
      }
      else {
        return (
          <Animated.View

            key={item.id} style={[{
              opacity: this.nextCardOpacity,
              transform: [{ scale: this.nextCardScale }],
              height: SCREEN_HEIGHT - 120, width: SCREEN_WIDTH, padding: 10, position: 'absolute'
            }]}>
           
            <LikeComponent style={{ likeAnimation: styles.likeAnimation, Like: styles.Like }} />
            <NopeComponent style={{ nopeAnimation: styles.nopeAnimation, Nope: styles.Nope }} />


            <Image
              style={styles.image}
              source={item.uri} />

          </Animated.View>
        )
      }
    }).reverse()
  }

  render() {
    return (
      <View style={{ flex: 1, }}>
  
        <View style={{ flex: 1 }}>
          {this.renderTech()}
          {this.state.currentIndex === Tech.length ? <EndComponent style={{ height: SCREEN_HEIGHT - 120, width: SCREEN_WIDTH, padding: 20, justifyContent: 'center', alignItems: 'center' }}/>: null}
        </View>
        <View style={{ height: 60 }}>

        </View>


      </View>

    );
  }
}

HomeScreen.navigationOptions = {
  title: 'Hot or Not',
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  imageContainer: {

  },
  image: {
    flex: 1, height: null,
    width: null,
    resizeMode: 'contain',
    borderRadius: 20,
    backgroundColor: 'white'
  },
  Nope: { borderWidth: 1, borderColor: 'red', color: 'red', fontSize: 32, fontWeight: '800', padding: 10 },
  nopeAnimation: { opacity: 0, transform: [{ rotate: '30deg' }], position: 'absolute', top: 50, right: 40, zIndex: 1000 },
  Like: { borderWidth: 1, borderColor: 'green', color: 'green', fontSize: 32, fontWeight: '800', padding: 10 },
  likeAnimation: { opacity: 0, transform: [{ rotate: '-30deg' }], position: 'absolute', top: 50, left: 40, zIndex: 1000 },
});