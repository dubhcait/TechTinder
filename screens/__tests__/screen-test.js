import React from 'react';
import renderer from 'react-test-renderer';

import HomeScreen from '../HomeScreen';
import LinkScreen from '../LinksScreen';



describe('<HomeScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<LinkScreen />`', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LinkScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});