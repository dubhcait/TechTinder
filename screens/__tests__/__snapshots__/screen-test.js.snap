// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 1214,
          "opacity": 0,
          "padding": 10,
          "position": "absolute",
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "height": 1214,
          "padding": 20,
          "position": "absolute",
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "left": 40,
            "opacity": 0,
            "position": "absolute",
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "-30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "green",
              "borderWidth": 1,
              "color": "green",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          LIKE
        </Text>
      </View>
      <View
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "right": 40,
            "top": 50,
            "transform": Array [
              Object {
                "rotate": "30deg",
              },
            ],
            "zIndex": 1000,
          }
        }
      >
        <Text
          style={
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "color": "red",
              "fontSize": 32,
              "fontWeight": "800",
              "padding": 10,
            }
          }
        >
          NOPE
        </Text>
      </View>
      <Image
        source={1}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 60,
      }
    }
  />
</View>
`;

exports[`<LinkScreen />\` renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 15,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "height": 140,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "resizeMode": "contain",
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        This is a personal project and just a bit of fun.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        However, if you would like create an issue on my repo to leave feedback, It would make my day.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
        The code for this app is avalible
      </Text>
      <Text
        onPress={[Function]}
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "800",
            "marginVertical": 20,
            "textAlign": "center",
          }
        }
      >
         here.
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
